flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -38,13 +40,21 @@ const _navigationKey = 'navigation';
 ///   - [RouteObserver](https://api.flutter.dev/flutter/widgets/RouteObserver-class.html)
 ///   - [Navigating with arguments](https://flutter.dev/docs/cookbook/navigation/navigate-with-arguments)
 class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
-  SentryNavigatorObserver({Hub? hub, bool setRouteNameAsTransaction = false})
+  SentryNavigatorObserver(
+      {Hub? hub,
+      bool enableAutoTransactions = true,
+      bool setRouteNameAsTransaction = false})
       : _hub = hub ?? HubAdapter(),
+        _enableAutoTransactions = enableAutoTransactions,
         _setRouteNameAsTransaction = setRouteNameAsTransaction;
 
   final Hub _hub;
+  final bool _enableAutoTransactions;
   final bool _setRouteNameAsTransaction;
 
+  ISentrySpan? _transaction;
+  Timer? _idleTimer;
```

Comment:
I believe this should be builtin in the `SentryTracer` class, since it the class that have access to its children, and should reset the timer when a child span is created

Reply:
But if the timer ends inside the SentryTracer it would not finish with status 'ok' right?

Reply:
With the move to `SentryTracer`, we can now also call finish to be done delayed with a duration and a status. The latter wil only be set if it is empty.